src/eval/func.rs

```diff
@@ -144,6 +145,31 @@ impl Func {
             _ => None,
         }
     }
+
+    /// Get a field from this function's scope,
+    /// if possible.
+    pub fn get(&self, field: &str) -> StrResult<&Value> {
+        match &self.repr {
+            Repr::Native(func) => func.info.scope.get(field).ok_or_else(|| {
+                eco_format!(
+                    "function `{}` does not contain field `{}`",
+                    func.info.name,
+                    field
+                )
+            }),
+            Repr::Elem(func) => func.info().scope.get(field).ok_or_else(|| {
+                eco_format!(
+                    "function `{}` does not contain field `{}`",
+                    func.name(),
+                    field
+                )
+            }),
+            Repr::Closure(_) => Err(eco_format!(
+                "cannot access fields on closures and user-defined functions"
+            )),
+            Repr::With(arc) => arc.0.get(field),
```

Comment:
does this make sense? this allows `assert.with(2).eq(..)`, but the `with` has no effect.

Reply:
I _personally_ think it's fine to accept this, mostly because I think people might often use templates which override default functions using `with`, and so the scoped functions could become inaccessible. (Of course, they do become inaccessible if people replace them with closures, which can be a problem... but this is the best we can do for now, I guess.)